projects/instant-ui-library/src/lib/loading-spinner/loading-spinner/loading-spinner.component.ts

```diff
@@ -0,0 +1,70 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  Input,
+  OnChanges,
+  SimpleChanges,
+} from '@angular/core';
+import { ModeType } from '../../shared/models/color.model';
+import { Size, SizeMode } from '../../shared/models/size.model';
+import { Mode } from './../../shared/models/color.model';
+
+@Component({
+  selector: 'iu-loading-spinner[id]',
+  templateUrl: './loading-spinner.component.html',
+  styleUrls: ['./loading-spinner.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class LoadingSpinnerComponent implements OnChanges {
+  @Input() variant: ModeType;
+  @Input() sizeMode: SizeMode;
+  @Input() id: string = '';
+
+  spinnerSizeClass: string = '';
+  spinnerColorClass: string = '';
+  size = Size;
+  mode = Mode;
+
+  ngOnChanges(changes: SimpleChanges) {
+    if (changes) {
```

Comment:
You can do a check for specific input changes, unless the intent is for all changes run the code.

Reply:
Noice.